### ISSUE TITLE
Expand Textfield control and options

### ISSUE THREAD

**Issue by renshenhe** [kradio3/react-mdc-web#24]
I may have missed something but it does not seem `Textfield` component allows control of input validation e.g.
```js
<Textfield
  value={this.props.username}
  helptext="Username contains illegal characters"
  />
```
The validation seems restricted to the element's built in validations rather than by the component.

`Textfield` component's `helptext` property has issues with the component's inner `this.focus` when passed dynamic values.  This causes inputs to shift focus.

I am unsure whether these aspects  were intended or a mistake on my part.  I am very unexperienced so please let me know of any issues on my part.

Keep up the great work!

**Comment by kradio3** [comment]
Hi @renshenhe , 
Sorry, i cannot understand the problem.
Could you please reproduce it in this Codepen https://codepen.io/kradio3/pen/jwbNqy ?

**Comment by renshenhe** [comment]
Sorry for the late response, actually slipped my mind.  Oddly the issue disappeared despite no changes in my code.  I had it run as an isolated component prior to reproduce the issue though I can't seem to reproduce it.  I have recently upgraded dependencies that may have previously caused hidden collision with the `Textfield` component. 

Closing since I cannot reproduce the issue anymore. Sorry for the trouble.

Would still hope to have control over helptext validation i.e. able to trigger validation error outside of native validations.
